cms/server/ContestWebServer.py

```diff
@@ -1061,10 +1064,12 @@ def which_language(user_filename):
                              if it is not a recognized language.
 
             """
-            for source_ext, language in SOURCE_EXT_TO_LANGUAGE_MAP.iteritems():
-                if user_filename.endswith(source_ext):
-                    return language
-            return None
+
+            languageSubmitted = self.get_argument("submission_lang")
```

Comment:
Wait, does this actually do The Right Thing now? O.o

I was under the impression get_argument will throw MissingArgumentError if the argument doesn't exist + no default is provided, so surely there's A Big Problem if this happens